ddl/table.go

```diff
@@ -208,7 +206,33 @@ func onTruncateTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
+	oldPid := getPartitionIDs(tblInfo)
 	tblInfo.ID = newTableID
```

Comment:
I suggest not to touch the old table Info here, before the new table info is ready.
If you change something and meet errors later, what will happen?

Reply:
@tiancaiamao This just saves the partitionID of the old table, no old table modified.

Reply:
Error handling is not that simple, please take a close look.

```
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)	
```